### PR TITLE
dynamic difficulty

### PR DESCRIPTION
**Pull request by tristanRW**

### CODE REVIEW

File: lib/miner/index.js [hunk 1]
```diff
@@ -1,182 +1,231 @@
...
+              inputTransactionsInTransaction(transactionsInBlocks),
+            ),
+          ),
+        );
+
+        return wasItFoundInSelectedTransactions || wasItFoundInBlocks;
+      }, transaction.data.inputs);
+
+      if (R.all(R.equals(false), transactionInputFoundAnywhere)) {
+        if (transaction.type === "regular" && negativeOutputsFound === 0) {
```
Comment (yikchunnnn) [inline lib/miner/index.js:129]: if ((transaction.type === 'regular' || transaction.type === 'create_event_wallet' || transaction.type === 'register') && negativeOutputsFound === 0) {

File: lib/config.js [hunk 2]
```diff
@@ -1,44 +1,85 @@
...
-    pow: {
-        getDifficulty: (blocks, index) => {
-            // Proof-of-work difficulty settings
-            const BASE_DIFFICULTY = Number.MAX_SAFE_INTEGER;
-            const EVERY_X_BLOCKS = 5;
-            const POW_CURVE = 5;
+  // INFO: The mining reward could decreases over time like bitcoin. See https://en.bitcoin.it/wiki/Mining#Reward.
+  MINING_REWARD: 5000000000,
+  // INFO: Usually it's a fee over transaction size (not quantity)
+  FEE_PER_TRANSACTION: 1,
```
Comment (yikchunnnn) [inline lib/config.js:6]: FEE_PER_TRANSACTION: 0,

**Review by yikchunnnn — CHANGES_REQUESTED** [review]
2 conflicts found

**Comment by tristanRW** [comment]
now?

**Review by yikchunnnn — APPROVED** [review]

**Pull request merged**